geth/node/manager.go

```diff
@@ -140,9 +141,10 @@ func (m *NodeManager) StopNode() (<-chan struct{}, error) {
 
 // stopNode stop Status node. Stopped node cannot be resumed.
 func (m *NodeManager) stopNode() (<-chan struct{}, error) {
-	if m.node == nil || m.nodeStarted == nil || m.nodeStopped == nil {
-		return nil, ErrNoRunningNode
+	if err := m.isNodeAvailable(); err != nil {
+		return nil, err
 	}
```

Comment:
1. These 2 conditions are not equivalent.
2. Also, I think this check should rather be in `m.StopNode`, right?

Reply:
1. Not trivial, but it looks it is. If `m.nodeStopped` is `nil`, then `m.node` must be `nil` as well. It's because `m.nodeStopped` is `nil` only before the node is started and is never set to `nil` afterwards. It gets its value in a lock right after `m.node` is assigned. Generally, this whole logic to start/stop a node is overcomplicated with locks and channels floating around. We can refactor this file and save 50% LOC.
2. Correct.

Reply:
Well, my concern is about this: https://github.com/status-im/status-go/blob/f445b1139465dcda19511da61a9211781a2bfbcc/geth/node/manager.go#L101
`nodeStopped`
First, `node` is assigned and only then `nodeStopped`. So if somebody calls `stopNode` in parallel with `startNode`, there may be a race condition that `node` is already assigned while `nodeStopped` is not yet.
The initial if condition would return true here while yours will return false.

Reply:
`node` and `nodeStopped` assignments are guarded with a `Lock()`. `StopNode()` also makes a `Lock()` so it's not possible. But I think we can just add this one condition, it won't hurt anything, it just will be an additional check.

Reply:
Well, yeah, now they are. But when you move `isNodeAvailable` to `StopNode` they won't be, right?

Reply:
They will, because `isNodeAvailable()` call will be after acquiring a lock. Anyway, I will add that condition.